### ISSUE TITLE
[Doc]: Release docs include Next what's new pages

### ISSUE THREAD

**Issue by dstansby** [matplotlib/matplotlib#27864]
### Documentation Link

https://matplotlib.org/stable/users/next_whats_new.html

### Problem

Ideally this page shouldn't be in the docs for releases.

**Comment by rcomer** [comment]
It's in the sidebar of https://matplotlib.org/stable/users/release_notes.html.  Is there a way to exclude it from there?  It wasn't there for v3.7.

**Comment by timhoffm** [comment]
The sidebar is now empty since #27800.

**Comment by dstansby** [comment]
The sidebar on e.g., https://matplotlib.org/devdocs/users/prev_whats_new/whats_new_3.8.0.html isn't empty though, and that shows up with "Next what's new?" and "Next API changes" at the top of the sidebar:
<img width="396" alt="Screenshot 2024-03-05 at 19 50 20" src="https://github.com/matplotlib/matplotlib/assets/6197628/c7d95266-637e-4392-b87b-5b5bcfbbf70f">

**Comment by timhoffm** [comment]
Yes, but #27800 deactivates that sidebar and is milestoned for 3.9.

**Comment by dstansby** [comment]
As far as I can tell, it only deactivates it for one page, https://matplotlib.org/devdocs/users/release_notes.html, and not other pages, e.g., https://matplotlib.org/devdocs/users/prev_whats_new/whats_new_3.8.0.html. The sidebar is turning up on that link, which is a devdocs link.

**Comment by timhoffm** [comment]
Oh, I see. Then, #27800 should be extended to show blank sidebars as well for `api/prev_api_changes/*` and `users/prev_whats_new/*`, which are logical subpages of the release notes.

On a side note, the relation between the logical toc hierarchy and the file structure is not quite trivial 😢.

**Comment by dstansby** [comment]
I think being on the "What's new for 3.8" and being able to see the table of contents is useful though, for example to go to the API changes or GitHub statistics. Surely the right solution to this is find a way to exclude "Next what's new" and "Next API changes" from release builds?

**Comment by timhoffm** [comment]
> I think being on the "What's new for 3.8" and being able to see the section navigation is useful though, for example to go to the API changes or GitHub statistics. 

I agree in theory, but due to the somewhat convoluted structure the menu is rather unusable. Note that there is a mix of 3.8.0 and 3.8.1 topics and "Github statistics for 3.8.1" even appears twice:

![grafik](https://github.com/matplotlib/matplotlib/assets/2836374/9bfb15b4-d02c-4516-9204-a4960b231bbc)

So until someone tries to untangle our incongruent use of logical hierarchy (sections) and page hierarchy, this side bar is rubbish.

**Comment by timhoffm** [comment]
Let's close this. While the empty release notes are still there https://matplotlib.org/stable/users/release_notes.html, the sidebar is cleaned up, and thus the existence of the empty page does not matter too much.
